.github/workflows/ci.yml

```diff
@@ -16,6 +16,12 @@ jobs:
           sudo apt-get update
           sudo apt-get install atlassian-plugin-sdk=8.2.8
           atlas-version
+      - name: Cache Maven packages
+        uses: actions/cache@v4
+        with:
+          path: ~/.m2
+          key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
+          restore-keys: ${{ runner.os }}-m2

```

Comment:
So if I understand correctly, even if the files have changed (`hashFiles` is different from a previous run) we can still end up restoring the cache using the restore key if we're on the same OS.

I assume that Maven is smart enough that it will never cause any issues? If a pom file has changed it'll rerun with no issues and we won't get any accidental skips?